parenthetics.py

```diff
@@ -0,0 +1,39 @@
+#! /usr/bin/env python
+# parenthetics - version using stack
+#
+# by Henry Grantham
+
+from __future__ import unicode_literals
+
+import stack
+
+
+def parenthetics(text):
+    """Determines if a string is "open", "balanced", or "broken" given a string.
+
+    Takes a unicode string (text) as input and returns one of
+    three possible values:
+    1 if the string is "open"
+        (there are open parens that are not closed)
+    0 if the string is "balanced"
+        (there are an equal number of matching open and closed parenthasis)
+    1 if the string is "broken"
+    (a closing parens has not been proceeded by one that opens)
+    """
+
+    # number of open parens
+    parens = stack.Stack()
+
+    text_list = list(text)
+
+    while text_list:
```

Comment:
Why build a list out of a string and then proceed to consume it like this?  Could you not just as well do:

```python
for element in text:
    #continue on as before
```